flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -1894,6 +1895,52 @@ object TaskManager {
       override def getValue: Double =
         ManagementFactory.getOperatingSystemMXBean().getSystemLoadAverage()
     })
+    // Preprocessing steps for registering cpuLoad
+    // fetch the method to get process CPU load
+    val getCPULoadMethod: Method = getMethodToFetchCPULoad()
+
+    // Log getProcessCpuLoad method not available for Java 6
+    if(getCPULoadMethod == null){
+      LOG.warn("getProcessCpuLoad method not available in the Operating System Bean" +
+      "implementation for this Java runtime environment\n"+Thread.currentThread().getStackTrace)
+    }
+
+    // define the fetchCPULoad method as per the fetched getCPULoadMethod
+    val fetchCPULoad: (Any) => Double = if (getCPULoadMethod != null) {
+      (obj: Any) => getCPULoadMethod.invoke(obj).asInstanceOf[Double]
+    } else {
+      (obj: Any) => -1
+    }
+
+    metricRegistry.register("cpuLoad", new Gauge[Double] {
+      override def getValue: Double = {
+        try{
```

Comment:
Why aren't you doing the check `getCPULoadMethod != null` before the cast?
if it is null, return -1, if not, do the cast and fetch the CPU load.